src/hostname/hostnamed.c

```diff
@@ -250,8 +250,27 @@ static int get_firmware_vendor(char **ret) {
          return get_hardware_firmware_data("bios_vendor", ret);
 }
 
-static int get_firmware_date(char **ret) {
-         return get_hardware_firmware_data("bios_date", ret);
+static int get_firmware_date(uint64_t *ret) {
+         const char *k;
+         struct tm tm;
+         time_t v;
+         _cleanup_free_ char *bios_date = NULL;
+         int r;
+
+         r = get_hardware_firmware_data("bios_date", &bios_date);
+         if (r <= 0)
+                 return r;
```

Comment:
Please set 'a default' value on `r == 0`:
```c
r = get_hardware_firmware_data(...);
if (r < 0)
        return r;
if (r == 0) {
        *ret = USEC_INFINITY;
        return 0;
}
```

Reply:
not addressed yet

Reply:
ah, it is addressed, but you forgot to replace the `<=` with `<` here